network/README.md

```diff
@@ -4,45 +4,45 @@ title: Network
+mempool, and state sync protocols. DiemNet tries to maintain at-most one connection
+with each remote peer; the application protocols to that remote peer are then
+multiplexed over the single peer connection.
 
-For more detailed info, see the [Network Specification](../specifications/network/README.md).
+Currently, it provides application protocols with two primary interfaces:
 
-The network component is specifically designed to facilitate the consensus and
-shared mempool protocols. Currently, it provides these consumers with two
-primary interfaces:
-* RPC, for Remote Procedure Calls; and
-* DirectSend, for fire-and-forget style message delivery to a single receiver.
+* DirectSend: for fire-and-forget style message delivery.
+* RPC: for unary Remote Procedure Calls.
 
 The network component uses:
+
 * TCP for reliable transport.
-* [Noise](https://noiseprotocol.org/noise.html) for authentication and full
- end-to-end encryption.
-* On-chain addresses for discovery, with a seed peers file for initial discovery.
-
-Each new substream is assigned a *protocol* supported by both the sender and
-the receiver. Each RPC and DirectSend type corresponds to one such protocol.
-
-Only eligible members are allowed to join the inter-validator network. Their
-identity and public key information is provided by the consensus
-component at initialization and on validator set reconfiguration. A new
-validator also needs the network addresses of a few *seed* peers to help it
-bootstrap connectivity to the network. The seed peers first authenticate the
-joining validator as an eligible member and then share their network state
-with it.
-
-Each member of the network maintains a full membership view and connects
-directly to any validator it needs to communicate with. A validator that cannot
-be connected to directly is assumed to fall in the quota of Byzantine faults
-tolerated by the system.
+* [NoiseIK](https://noiseprotocol.org/noise.html) for authentication and full
+  end-to-end encryption.
+* On-chain [`NetworkAddress`](./network-address/src/lib.rs) set for discovery, with
+  optional seed peers in the [`NetworkConfig`](../config/src/config/network_config.rs)
+  as a fallback.
+
+Validators will only allow connections from other validators. Their identity and
+public key information is provided by the [`simple-onchain-discovery`] protocol,
+which updates the eligible member information on each consensus reconfiguration.
+
+In contrast, Validator Full Node (VFNs) servers will only prioritize connections
+from more trusted peers in the on-chain discovery set; they will still service
+any public clients. Public Full Nodes (PFNs) connecting to VFNs will always
+authenticate the VFN server using the available discovery information.
+
+Each member of the validator network maintains a full membership view and connects
+directly to all other validators in order to maintain a full-mesh network.
```

Comment:
Move this up next to the `Validator` section to keep it together with the other validator comments